### PR TITLE
fix: Ignore bin dependencies in lib

### PR DESCRIPTION
**Pull request by taorepoara**
## Checklist
- [x] I didn't over-scope my PR
- [x] My PR title matches the [commit convention](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I did not include breaking changes
- [ ] I included unit tests that cover my changes
- [x] I added/updated the documentation about my changes
- [x] I made my own code-review before requesting one

## Description of the changes
Do not load the bin dependencies in the lib crate.

## Technical highlight/advice

The `cargo install` must specify the bin feature: `cargo install dofigen -F cli`

### CODE REVIEW

**Comment by taorepoara** [comment]
@jrouaix What do you think ?

**Comment by jrouaix** [comment]
I wonder why `-F` is need when `cargo install` command.
The install command should only consider the one `bin` part of the crate so is should activate the feature itself ?
(perhaps i'm mistaken here and the required_feature does not activate the feature automaticaly ...)

**Comment by taorepoara** [comment]
> I wonder why `-F` is need when `cargo install` command.
> The install command should only consider the one `bin` part of the crate so is should activate the feature itself ?
> (perhaps i'm mistaken here and the required_feature does not activate the feature automaticaly ...)

I have tried without it but the install fails because the feature is not activated.
I also tried with the `bin` option (`cargo install --path . --bin dofigen`) but the installation also fails and says explicitly that the `--features bin` option is needed because of the `required_feature`.

File: Cargo.toml [hunk 1]
```diff
@@ -20,9 +20,9 @@ path = "src/lib.rs"
 [[bin]]
 name = "dofigen"
 path = "src/bin/main.rs"
+required-features = ["bin"]
```
Comment (shiipou) [inline Cargo.toml:23]: The feature `bin` didn't give enough meaning to final user,
I think naming it `cli` can help people to know the feature it to add the command line features.

```suggestion
required-features = ["cli"]
```

**Review by shiipou — CHANGES_REQUESTED** [review]

File: .github/workflows/build_ci.yml [hunk 2]
```diff
@@ -158,7 +158,7 @@ jobs:
       with:
         use-cross: true
         command: build
-        args: --release --target ${{ matrix.target }}
+        args: --release -F bin --target ${{ matrix.target }}
```
Comment (shiipou) [inline .github/workflows/build_ci.yml:161]:
```suggestion
        args: --release -F cli --target ${{ matrix.target }}
```

File: dofigen.yml [hunk 3]
```diff
@@ -6,7 +6,7 @@ builders:
   - "."
   script:
   # Build with musl to work with scratch
-  - cargo build --release --target=x86_64-unknown-linux-musl
+  - cargo build --release -F bin --target=x86_64-unknown-linux-musl
```
Comment (shiipou) [inline dofigen.yml:9]:
```suggestion
  - cargo build --release -F cli --target=x86_64-unknown-linux-musl
```

**Review by shiipou — CHANGES_REQUESTED** [review]
I think you missed this `bin` features

**Review by shiipou — APPROVED** [review]
LGTM

**Pull request merged**